src/agenda/Contacto.java

```diff
@@ -25,13 +19,46 @@ public Contacto(String nombre, String apellido, String email, String movil){
 		this.movil=movil;
 	}
 
-//Funcion para seteo de Nombre.
-	public void setNombre(String nomb){
-		this.nombre=nomb.toUpperCase();
+	public void setNombre(String nombre){
+		this.nombre=nombre.toUpperCase();
+	}
+	
+	public String getNombre() {
+		return nombre;
+	}
+	
+	public void setApellido(String apellido) {
+		this.apellido = apellido;
+	}
+	
+	public String getApellido() {
+		return apellido;
+	}
+	
+	public void setEmail(String email) {
+		this.email = email;
 	}
 	
 	public String getEmail(){
-		return this.email;
+		return email;
+	}
+
+	public void setMovil(String movil) {
+		this.movil = movil;
+	}
+
+	public String getMovil() {
+		return movil;
 	}
 	
+@Override
+    public boolean equals(Object obj) {
+        if (obj == null) return false;
+        if (getClass() != obj.getClass()) return false;
+        final Contacto other = (Contacto) obj;
+        boolean sameA = (this.email == other.email) || (this.email != null && this.email.equalsIgnoreCase(other.email));
```

Comment:
Bien, ese es el principal refactor: quitar el if innecesario.
En segundo lugar, podes aplicar el refactor "inline", lo haces desde la IDE (boton derecho sobre la variable), y lo que hace es utilizar el valor que le asignas a la variable para usarlo directamente en lugar de su nombre. Te quedaría algo así:

return (this.email == other.email) || (this.email != null && this.email.equalsIgnoreCase(other.email));